docs/core-features/table_lookups_basics.md

```diff
@@ -0,0 +1,186 @@
+# Table lookup basics
+
+In TFHE, there exists mainly two operations: the linear operations (additions, subtractions, multiplications by integers) and the rest. And the rest is done with table lookups (TLUs), which means that a lot of things are done with TLU. In this document, we explain briefly, from a user point of view, how TLU can be used. In [the poweruser documentation](../dev/compilation/compiler_internals.md), we enter a bit more into the details.
+
+## Direct TLU
+
+Direct TLU stands for instructions of the form `y = T[i]`, for some table `T` and some index `i`. One can use the `fhe.LookupTable` to define the table values, and then use it on scalars or tensors.
+
+```python
+from concrete import fhe
+
+table = fhe.LookupTable([2, -1, 3, 0])
+
+@fhe.compiler({"x": "encrypted"})
+def f(x):
+    return table[x]
+
+inputset = range(4)
+circuit = f.compile(inputset)
+
+assert circuit.encrypt_run_decrypt(0) == table[0] == 2
+assert circuit.encrypt_run_decrypt(1) == table[1] == -1
+assert circuit.encrypt_run_decrypt(2) == table[2] == 3
+assert circuit.encrypt_run_decrypt(3) == table[3] == 0
+```
+
+```python
+from concrete import fhe
+import numpy as np
+
+table = fhe.LookupTable([2, -1, 3, 0])
+
+
+@fhe.compiler({"x": "encrypted"})
+def f(x):
+    return table[x]
+
+
+inputset = [np.random.randint(0, 4, size=(2, 3)) for _ in range(10)]
+circuit = f.compile(inputset)
+
+sample = [
+    [0, 1, 3],
+    [2, 3, 1],
+]
+expected_output = [
+    [2, -1, 0],
+    [3, 0, -1],
+]
+actual_output = circuit.encrypt_run_decrypt(np.array(sample))
+
+assert np.array_equal(actual_output, expected_output)
+```
+
+`LookupTable` mimics array indexing in Python, which means if the lookup variable is negative, the table is looked up from the back.
+
+## Indirect TLU
```

Comment:
We never use this term anywhere. Why not `## Multi TLU`?

Reply:
I would have said that the name of this in programming langage is Indirect TLU, but sure, I can move to Multi TLU

Reply:
done

Reply:
Yes indirect TLU could be a name, but we never expose the indirection (second index) to the users. It's just an operation to apply multiple different tables to cells of a tensor, hence multi tlu.

Reply:
Indeed! I guess it could be an option for future, and could be useful, actually :)